core/src/main/java/org/eclipse/hono/util/RequestResponseApiConstants.java

```diff
@@ -0,0 +1,198 @@
+/**
+ * Copyright (c) 2017 Bosch Software Innovations GmbH.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial creation
+ */
+package org.eclipse.hono.util;
+
+import io.vertx.core.json.JsonObject;
+import io.vertx.proton.ProtonHelper;
+import org.apache.qpid.proton.amqp.Symbol;
+import org.apache.qpid.proton.amqp.messaging.AmqpValue;
+import org.apache.qpid.proton.amqp.messaging.ApplicationProperties;
+import org.apache.qpid.proton.amqp.messaging.MessageAnnotations;
+import org.apache.qpid.proton.message.Message;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static org.eclipse.hono.util.MessageHelper.decodeIdFromJson;
+
+/**
+ * Constants &amp; utility methods that are common to APIs that follow the request response pattern.
+ */
+public class RequestResponseApiConstants {
+    /* message payload fields */
+    public static final String FIELD_PAYLOAD                     = "payload";
+    public static final String FIELD_ENABLED                     = "enabled";
+
+    /* message property names */
+    public static final String APP_PROPERTY_KEY                  = "key";
+
+    /**
+     * Build a Proton message as a reply for an endpoint from the json message received from the vert.x eventbus
+     * from the implementing service.
+     *
+     * @param endpoint The endpoint the reply message will be built for.
+     * @param message The json message received.
+     * @return Message The built Proton message.
+     */
+    public static final Message getAmqpReply(final String endpoint, final io.vertx.core.eventbus.Message<JsonObject> message) {
+        final String tenantId = message.body().getString(MessageHelper.APP_PROPERTY_TENANT_ID);
+        final String deviceId = message.body().getString(MessageHelper.APP_PROPERTY_DEVICE_ID);
+        final String status = message.body().getString(MessageHelper.APP_PROPERTY_STATUS);
+        final JsonObject correlationIdJson = message.body().getJsonObject(MessageHelper.SYS_PROPERTY_CORRELATION_ID);
+        final Object correlationId = decodeIdFromJson(correlationIdJson);
+        final boolean isApplCorrelationId = message.body().getBoolean(MessageHelper.ANNOTATION_X_OPT_APP_CORRELATION_ID, false);
+        return getAmqpReply(endpoint, status, correlationId, tenantId, deviceId, isApplCorrelationId,
+                message.body().getJsonObject(CredentialsConstants.FIELD_PAYLOAD));
+    }
+
+    /**
+     * Build a Proton message as a reply for an endpoint from the json message received from the vert.x eventbus
```

Comment:
that is not true, this method takes the individual values that have been conveyed in the message received via the event bus. So this should probably be something like:
> Creates an AMQP message containing the reply to a request.